pandas/tests/indexes/multi/test_unique_and_duplicates.py

```diff
@@ -122,14 +121,26 @@ def test_duplicate_meta_data():
         assert idx.drop_duplicates().names == idx.names
 
 
-def test_duplicates(idx):
-    assert not idx.has_duplicates
-    assert idx.append(idx).has_duplicates
+def test_has_duplicates():
+    major_axis = pd.Index(['foo', 'bar', 'baz', 'qux'])
+    minor_axis = pd.Index(['one', 'two'])
 
-    index = MultiIndex(levels=[[0, 1], [0, 1, 2]], labels=[
-        [0, 0, 0, 0, 1, 1, 1], [0, 1, 2, 0, 0, 1, 2]])
+    major_labels = np.array([0, 0, 1, 2, 3, 3])
+    minor_labels = np.array([0, 1, 0, 1, 0, 1])
+    index_names = ['first', 'second']
+    mi = pd.MultiIndex(levels=[major_axis, minor_axis],
+                       labels=[major_labels, minor_labels],
+                       names=index_names, verify_integrity=False)
+    assert not mi.has_duplicates
+    assert mi.append(mi).has_duplicates
+
+    index = pd.MultiIndex(levels=[[0, 1], [0, 1, 2]],
```

Comment:
This check seems a little "imbalanced" compared to what's above. Can we simplify this test through parametrization instead?

Reply:
I don't understand what unbalanced is supposed to mean here. It is exactly how this test worked pre #21514.